pkg/security/security.go

```diff
@@ -351,12 +352,36 @@ type Caller struct {
 	Identities []string
 }
 
+type AuthContext struct {
+	// RequestContext is the context from request.
+	RequestContext context.Context
+	// Authenticators is the list of authenticators that were executed before in the auth chain.
+	Authenticators []string
+	// DelegatedAuthenticators are the list of delegated authenticators that main authenticators can
+	// choose to get identities from. For example, CidrAuthenticator can delegate the authentication to
+	// XfccAuthenticator once it validates the connection is from trusted cidr range. XfccAuthenticator
+	// can then extract identities from peer certificate.
+	DelegatedAuthenticators []Authenticator
```

Comment:
Do we need this? It seems to make the calling complex. I was thinking we would just pass in a `Identities []string` as part of the inputs, which is the "currently known identities".  Or I guess if we are doing CIDR auth instead of based on the callers real identity, then just some bool that says "trustednetwork" or whatever and XFCC reads that variable.

Reply:
I am confused even with the comments

Reply:
This is not just for XFCC case. We could add other delegated authenticators later if needed. For XFCC though, having the bool may be sufficient.